app/models/journey.rb

```diff
@@ -0,0 +1,42 @@
+class Journey < ApplicationRecord
+  belongs_to :move
+  belongs_to :supplier
+  belongs_to :from_location, class_name: 'Location'
+  belongs_to :to_location, class_name: 'Location'
+
+  enum states: {
+    in_progress: 'in_progress',
+    completed: 'completed',
+    cancelled: 'cancelled',
+  }
+
+  validates :move, presence: true
+  validates :supplier, presence: true
+  validates :from_location, presence: true
+  validates :to_location, presence: true
+  validates :client_timestamp, presence: true
+  validates :billable, exclusion: { in: [nil] }
+  validates :state, presence: true, inclusion: { in: states }
+
+  scope :default_order, -> { order(client_timestamp: :asc) }
+
+  delegate :complete, :uncomplete, :cancel, :uncancel, to: :state_machine
+
+  after_initialize :synchronise_state # NB there is an equivalent after(:build) callback used by FactoryBot in the journeys factory
```

Comment:
Call backs are difficult to maintain and make things difficult to understand (a step towards spaghetti code)
but I am not sure about the alternative atm, 
Any idea from everyone?

Reply:
I'm afraid that the callbacks are necessary to initialise the state_machine for the various events in a model's lifetime (`new()`, `build()`, `find()` and `create()`).